src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,346 @@
+    }
+
+    private static List<String> createWrongPhoneNumbers() {
+        List<String> wrongPhoneNumbers = getCorrectPhoneNumbers();
+        List<String> tempList = new ArrayList<>();
+        for (String str : wrongPhoneNumbers) {
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, " $2$3$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "+3$2$3$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "+ 3$2$3$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "+$2$3$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "3$2$3$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$212$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2 12$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2 12 $4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2+12$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2(12)$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2( 12)$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2(+12)$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2)$3$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2$3($4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2)$3($4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2$3$4($5)$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2$3$4$5($6)$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "+ 38$2$3$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "++38$2$3$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "+3 8$2$3$4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2 0 50 $4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2 05 0 $4$5$6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2$3$4 67 8 $6$7$8$9"));
+            tempList.add(str.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN, "$1$2$3$4 6 7 8 $6$7$8$9"));
+        }
+
+        List<String> specialChars = getSpecialCharacters("\\");
+        specialChars.removeAll(List.of("\\.", "\\-"));
+
+        specialChars.forEach(
+                specialChar -> wrongPhoneNumbers.forEach(
+                        number -> {
+                            tempList.add(
+                                    number.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN,
```

Comment:
Such logic I've already seen at the beginning of the method 198 - 221 lines and 177 - 187 lines as well. Extract it into a new method.